building/build-debs/sources-shared/src/keycommon/endpoint/endpoint.go

```diff
@@ -19,6 +19,23 @@ type ServerEndpoint struct {
 	extraHeaders map[string]string
 	certificates []tls.Certificate
 	timeout      time.Duration
+	client       *http.Client
+}
+
+func (s ServerEndpoint) buildClient() *http.Client {
```

Comment:
Could this function just modify ServerEndpoint's client directly, since that's how it's used in all instances?

Reply:
Probably, though then ServerEndpoint wouldn't always be passed by value.